api/openapi-spec.yaml

```diff
@@ -0,0 +1,473 @@
+openapi: 3.0.1
+info:
+  title: Event Repository REST API
+  description: Event Repository REST API for retrieving the event information
+  contact: {}
+  version: 0.0.1
+tags:
+- name: event-resource
+  description: The Event Resource API for getting single event information
+- name: search-resource
+  description: The search api provides methods for querying up/down stream given an
+    event id.
+- name: events-resource
+  description: The Events Resource API for getting all events information
+paths:
+  /events:
+    get:
+      tags:
+      - events-resource
+      summary: To get all events information
+      operationId: getEventsUsingGET
+      parameters:
+      - name: pageNo
+        in: query
+        description: "Page to display if results span across multiple pages."
+        schema:
+          type: integer
+          format: int32
+          default: 1
+      - name: pageSize
+        in: query
+        description: "The number of events to be displayed per page."
+        schema:
+          type: integer
+          format: int32
+          default: 500
+      - name: pageStartItem
+        in: query
+        description: "Intended to skip few items at the start of result, Should\
+          \ be used only if `pageNo=1` ie. first page of results."
+        schema:
+          type: integer
+          format: int32
+          default: 1
+      - name: shallow
+        in: query
+        description: "Determines if external ER's should be used to compile\
+          \ the results of query. Use `false` to use External ER's."
+        schema:
+          type: boolean
+          default: false
+      - name: lazy
+        in: query
+        description: "If lazy is `true`, it implies that when the events limit\
+          \ is reached according to pazesize no additional request is performed and the search will stop."
+        schema:
+          type: boolean
+          default: false
+      - name: readable
+        in: query
+        description: |
+          Determines if event time should be in milliseconds, or in Human readable time format. Ex:
+
+          `false` 1499076742982
+
+          `true`  2018-10-31T13:36:00.824Z.
```

Comment:
In both cases: Instead of

```json
      "meta": {
        "type": "EiffelArtifactCreatedEvent",
        "version": "3.0.0",
        "time": 1625989024288,
```

you would have

```json
      "meta": {
        "type": "EiffelArtifactCreatedEvent",
        "version": "3.0.0",
        "time": "2021-07-11T07:37:04.288",
```

Should I include examples to make it more clear?

Reply:
Oh, ouch. Do we really want to be changing the result payload? It's totally legit to adapt the results when they're rendered to a user but doing so in the API response seems suspect.

Reply:
I removed this option in a9debeb